File: Problems/26.remove-duplicates-from-sorted-array.py
#
# @lc app=leetcode id=26 lang=python3
#
# [26] Remove Duplicates from Sorted Array
#

# @lc code=start
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 0
        for j in nums:
            if nums[i] != j:
                i += 1
                nums[i] = j
        return i + 1

    def removeDuplicates2(self, nums: List[int]) -> int:
        # Iterate through all elements in nums
        i = 0
        while i < len(nums):
            # Check if current number is in nums (excluding itself)
            if nums[i] in nums[:i] + nums[i+1:]:
                del nums[i]
                i -= 1
            i += 1
        return len(nums)

# @lc code=end

File: leetcode/Problems/26.remove-duplicates-from-sorted-array.java
/*
 * @lc app=leetcode id=26 lang=java
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
class Solution {
    public int removeDuplicates(int[] nums) {
        // Two pointers for copying and pasting (scan is in for loop)
        int copyPos = 0;
        for (int scan = 0; scan < nums.length; scan++) {
            // Check if the two are now unequal
            if (nums[copyPos] != nums[scan]) {
                nums[++copyPos] = nums[scan];
            }
        }
        // Return the length
        return copyPos + 1;
    }
}
// @lc code=end

